operator/main.go

```diff
@@ -62,27 +76,17 @@ func init() {
 
 func main() {
 	var metricsAddr string
+
 	var enableLeaderElection bool
+
 	var probeAddr string
+
 	var maxConcurrentReconciles int
+
 	var requeueSuccessInterval, requeueFailureInterval, requeueWaitingInterval time.Duration
-	flag.StringVar(&metricsAddr, "metrics-bind-address", ":8080", "The address the metric endpoint binds to.")
-	flag.StringVar(&probeAddr, "health-probe-bind-address", ":8081", "The address the probe endpoint binds to.")
-	flag.IntVar(&maxConcurrentReconciles, "max-concurrent-reconciles", 1, "The maximum number of concurrent Reconciles which can be run.")
-	flag.BoolVar(&enableLeaderElection, "leader-elect", false,
-		"Enable leader election for controller manager. "+
-			"Enabling this will ensure there is only one active controller manager.")
 
-	flag.DurationVar(&requeueSuccessInterval, "requeue-success-interval", 20*time.Second,
-		"determines the duration after which an already successfully reconciled Kyma is enqueued for checking "+
-			"if it's still in a consistent state.")
-	flag.DurationVar(&requeueFailureInterval, "requeue-failure-interval", 10*time.Second,
-		"determines the duration after which a failing reconciliation is retried and "+
-			"enqueued for a next try at recovering (e.g. because an Remote Synchronization Interaction failed)")
-	flag.DurationVar(&requeueWaitingInterval, "requeue-waiting-interval", 3*time.Second,
-		"etermines the duration after which a pending reconciliation is requeued "+
-			"if the operator decides that it needs to wait for a certain state to update before it can proceed "+
-			"(e.g. because of pending finalizers in the deletion process)")
+	defineFlag(metricsAddr, probeAddr, maxConcurrentReconciles,
```

Comment:
If you want to set it up like this you can just create the variables inside the function instead of passing pointers!

Reply:
But how to use them? return all of them back?

Reply:
yes exactly, just define a struct with all the config values like on the old reconciler and then translate the config struct into the Reconciler struct by manually assigning values

Reply:
oh, how stupid I am.. I'm sometimes a bit doubt myself the capability of my code skill now, shouldn't relay too much on IDE auto refactoring in the future..